modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/file_id_query.rb

```diff
@@ -0,0 +1,45 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages::Peripherals::StorageInteraction::Nextcloud
+  class FileIdQuery
+    def initialize(storage)
+      @query = ::Storages::Peripherals::StorageInteraction::Nextcloud::Internal::PropfindQuery.new(storage)
+    end
+
+    def call(path:)
+      query_params = {
```

Comment:
Please define "entity". For me an entity is a data object in persistence.

If you mean, having multiple slightly different queries. It's because of in-code visibility. I want to make them distinguishable by name, not by a parameter that might be left out.

We can try to find a more elegant solution while keeping the explicit names intact though.

Reply:
I was using the term loosely here. Most meaning Object, from a behaviour standpoint both do the same thing, return a list of `file_info`. It just happens that one is a list consisting of one Object and the other possibly a longer list (but can also be just one Object).

Is it worth to have 2 different interfaces? Or have an unified one differing by what we are telling it to do?

Reply:
No, you are right, having the one with arrays is enough.